signer/src/stacks/api.rs

```diff
@@ -297,23 +295,16 @@ pub struct StacksClient {
 impl StacksClient {
     /// Create a new instance of the Stacks client using the given
     /// StacksSettings.
-    pub fn new(settings: StacksSettings) -> Self {
-        Self {
-            node_endpoints: settings.node.endpoints,
-            endpoint_index: 0,
-            nakamoto_start_height: settings.node.nakamoto_start_height,
-            client: reqwest::Client::new(),
-        }
-    }
-
-    /// iterate the stored endpoint index, wrapping
-    pub fn next_endpoint(&mut self) {
-        self.endpoint_index = (self.endpoint_index + 1) % self.node_endpoints.len();
-    }
+    pub fn new(url: Url, nakamoto_start_height: u64) -> Result<Self, Error> {
+        let client = reqwest::Client::builder()
+            .timeout(REQUEST_TIMEOUT)
+            .build()?;
```

Comment:
Oh interesting. Calling `reqwest::Client::new()` would panic if a TLS backend cannot be initialized, or if the resolver cannot load the system configuration. Using the builder bubbles up that same error. I was just surprised at the need for a `Result` here, figured it was good to share, no changes needed.